packages/vega-parser/src/parsers/guides/axis-ticks.js

```diff
@@ -38,14 +40,23 @@ export default function(spec, config, userEncode, dataRef, size, band) {
     round:  _('tickRound')
   };
 
-  if (orient === Top || orient === Bottom) {
-    update.y = enter.y = zero;
-    update.y2 = enter.y2 = tickSize;
-    update.x = enter.x = exit.x = tickPos;
+  if (isSignal(orient)) {
+    update.y = enter.y = xyAxisConditionalEncoding('x', orient.signal, zero, tickPos);
```

Comment:
It feels weird to have to parameter as 'x' for `enter.y`, same for the rest.

Reply:
(You can flip the yes and no case?)

Reply:
It makes more sense to me to read this as: the `y` position of the tick is at `zero` if we are dealing with the `x` axis. In that sense it seems unnecessary to force these values to be identical. 

I rewrote this too to make the logic more immediately similar to the non-signal case. I think it's easier to read these lines if the parameter is identical for all the conditionals (s.t. they can be grouped as `x` axis logic or `y` axis logic).